tests/model/operation/transform/attribute.js

```diff
@@ -0,0 +1,877 @@
+
+                syncClients();
+				expectClients( '<paragraph><$text bold="true">Foo</$text> <$text italic="true">Bar</$text></paragraph>' );
+
+                john.undo();
+
+				syncClients();
+                expectClients( '<paragraph>Foo <$text italic="true">Bar</$text></paragraph>' );
+            } );
+
+			it( 'in text in different path', () => {
+				john.setData( '<paragraph>F[o]o</paragraph><paragraph>Bar</paragraph>' );
+				kate.setData( '<paragraph>Foo</paragraph><paragraph>B[a]r</paragraph>' );
+
+				john.setAttribute( 'bold', true );
+				kate.setAttribute( 'italic', true );
+
+				syncClients();
+
+				expectClients(
+					'<paragraph>F<$text bold="true">o</$text>o</paragraph>' +
+					'<paragraph>B<$text italic="true">a</$text>r</paragraph>'
+				);
+			} );
+
+			it( 'in text with selection inside other client\'s selection #1', () => {
+				john.setData( '<paragraph>[Foo Bar]</paragraph>' );
+				kate.setData( '<paragraph>Fo[o B]ar</paragraph>' );
+
+				john.setAttribute( 'bold', true );
+				kate.setAttribute( 'italic', true );
+
+				syncClients();
+
+				expectClients(
+					'<paragraph>' +
+						'<$text bold="true">Fo</$text><$text bold="true" italic="true">o B</$text><$text bold="true">ar</$text>' +
+					'</paragraph>'
+				);
+			} );
+
+			it( 'in text with selection inside other client\'s selection #2', () => {
+				john.setData( '<paragraph>F[oo Ba]r</paragraph>' );
+				kate.setData( '<paragraph>Foo [Bar]</paragraph>' );
+
+				john.setAttribute( 'bold', true );
+				kate.setAttribute( 'italic', true );
+
+				syncClients();
+
+				expectClients(
+					'<paragraph>' +
+						'F' +
+						'<$text bold="true">oo </$text>' +
+						'<$text bold="true" italic="true">Ba</$text>' +
+						'<$text italic="true">r</$text>' +
+					'</paragraph>'
+				);
+			} );
+
+			it( 'in text with selection inside other client\'s selection #3', () => {
+				john.setData( '<paragraph><$text bold="true">[Foo Bar]</$text></paragraph>' );
+				kate.setData( '<paragraph><$text bold="true">Fo[o] Bar</$text></paragraph>' );
+
+				john.setAttribute( 'italic', true );
+				kate.setAttribute( 'underline', true );
+
+				syncClients();
+
+				expectClients(
+					'<paragraph>' +
+						'<$text bold="true" italic="true">Fo</$text>' +
+						'<$text bold="true" italic="true" underline="true">o</$text>' +
+						'<$text bold="true" italic="true"> Bar</$text>' +
+					'</paragraph>'
+				);
+			} );
+
+			it( 'in text at same position', () => {
+				john.setData( '<paragraph>[Foo Bar]</paragraph>' );
+				kate.setData( '<paragraph>[Foo Bar]</paragraph>' );
+
+				john.setAttribute( 'bold', true );
+				kate.setAttribute( 'italic', true );
+
+				syncClients();
+
+				expectClients( '<paragraph><$text bold="true" italic="true">Foo Bar</$text></paragraph>' );
+			} );
+
+			it( 'in element at same position', () => {
+				john.setData( '[<paragraph>Foo Bar</paragraph>]' );
+				kate.setData( '[<paragraph>Foo Bar</paragraph>]' );
```

Comment:
I am not sure about tests where the selection is around the paragraph. Is it possible to have such selection in the real case?

Reply:
Selection is not a case here. It is used only to not create ranges by hand.

Reply:
My point is, that it is very rare that we have a selection arount the block, it is more ofter to have a selection around texts, and tests should focus on such case, but I see more tests with `[<paragraph>Foo Bar</paragraph>]` than with `<paragraph>[Foo Bar]</paragraph>`.

Reply:
As I said it is not about the selection. What is needed is the range around `<paragraph>` not setting selection there. Selection is not even checked in the output data.